Doppler.Push.Api/Services/FirebaseCloudMessageService.cs

```diff
@@ -0,0 +1,44 @@
+using Doppler.Push.Api.Contract;
+using FirebaseAdmin.Messaging;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace Doppler.Push.Api.Services
+{
+    public class FirebaseCloudMessageService : IFirebaseCloudMessageService
+    {
+        public async Task<MessageSendResponse> SendMulticast(MessageSendRequest request)
+        {
+
+            var message = new MulticastMessage()
+            {
+                Notification = new Notification()
+                {
+                    Title = request.NotificationTitle,
+                    Body = request.NotificationBody,
+                },
+                Tokens = request.Tokens,
+            };
+
+            var response = await FirebaseMessaging.DefaultInstance.SendMulticastAsync(message);
+
+            var returnResponse = new MessageSendResponse()
+            {
+                Responses = response.Responses.Select(x => new ResponseItem
+                {
+                    IsSuccess = x.IsSuccess,
+                    MessageId = x.MessageId,
+                    Exception = x.IsSuccess ? null : new ExceptionItem
+                    {
+                        Message = x.Exception.Message,
+                        MessagingErrorCode = (int)x.Exception.MessagingErrorCode
```

Comment:
No, I was wrong!

```suggestion
                        MessagingErrorCode = (int?)x.Exception.MessagingErrorCode
```

or

```suggestion
                        MessagingErrorCode = x.Exception.MessagingErrorCode.HasValue ? (int)x.Exception.MessagingErrorCode : 0;
```

Because `MessagingErrorCode` can be null!

```
        //     Gets the Cloud Messaging error code associated with this exception. May be null.
        public MessagingErrorCode? MessagingErrorCode { get; }
```

Reply:
ok, lo tomo para ponerlo con esa opción, igualmente cuando es IsSuccess false en la respuesta de Firebase la enumeración no es null, el summary en el código fuente avisa que podria pero google en la documentación del SDK indica que no igualmente no esta de mas hacerlo bien y evitar problemas. Gracias!